mods/maintain-flux-colour-temperature.wh.cpp

```diff
@@ -0,0 +1,349 @@
+
+// Source code is published under The GNU General Public License v3.0.
+//
+// For bug reports and feature requests, please open an issue here:
+// https://github.com/levitation-opensource/my-windhawk-mods/issues
+//
+// For pull requests, development takes place here:
+// https://github.com/levitation-opensource/my-windhawk-mods/
+
+// ==WindhawkModReadme==
+/*
+# Maintain colour temperature of f.lux
+
+Keeps your preferred f.lux colour temperature settings, eliminating automatic changes. You can choose the desired colour temperature from f.lux dropdown menu, and f.lux will not change it later automatically.
+
+The mod operates by providing f.lux always with a fixed day and time. This fixed day and time can be adjusted by the user in the settings panel of this mod, if necessary, though the default setting should be usually suitable. By default the noon of approximately midsummer day of Northern Hemisphere is chosen as the fixed day and time. The particular day and time are not very important, what matters is that the fixed time is not near the time points where f.lux would usually get triggered to adjust the screen colour on its own.
+
+There are many posts in f.lux forum asking for this functionality since 2016. Unfortunately there were no solutions provided so far:
+* [Disable f.lux's automatic adjustments based on time/location](https://forum.justgetflux.com/topic/3012/disable-f-lux-s-automatic-adjustments-based-on-time-location/)
+* [Can automatic timer settings be disabled?](https://forum.justgetflux.com/topic/6162/can-automatic-timer-settings-be-disabled)
+* [Disabling all automatic functions.](https://forum.justgetflux.com/topic/6228/disabling-all-automatic-functions)
+* [How To Disable Automatic Change Based On Time / Location ?](https://forum.justgetflux.com/topic/7904/how-to-disable-automatic-change-based-on-time-location)
+* [How do I stop flux to stop changing colors arbitrary / Force always on mode / Force coupling day and night colors?](https://forum.justgetflux.com/topic/8404/how-do-i-stop-flux-to-stop-changing-colors-arbitrary-force-always-on-mode-force-coupling-day-and-night-colors)
+* [Sleeping during the day](https://forum.justgetflux.com/topic/8431/sleeping-during-the-day)
+* [How do I make my own Permanent preset?](https://forum.justgetflux.com/topic/8475/how-do-i-make-my-own-permanent-preset)
+* [Disable Circadian Response Only](https://forum.justgetflux.com/topic/8502/disable-circadian-response-only/)
+*/
+// ==/WindhawkModReadme==
+
+// ==WindhawkModSettings==
+/*
+- Hour: 12
+  $name: Hour in local time
+- Day: 21
+  $name: Day
+- Month: 6
+  $name: Month
+*/
+// ==/WindhawkModSettings==
+
+
+#include <windowsx.h>
+
+
+#ifndef WH_MOD
+#define WH_MOD
+#include <mods_api.h>
+#endif
+
+
+template <typename T>
+BOOL Wh_SetFunctionHookT(
+    FARPROC targetFunction,
+    T hookFunction,
+    T* originalFunction
+) {
+    return Wh_SetFunctionHook((void*)targetFunction, (void*)hookFunction, (void**)originalFunction);
+}

```

Comment:
`Wh_SetFunctionHookT` is already part of Windhawk, and can be used by including `windhawk_utils.h`.

Reply:
Thank you for including in the Windhawk, this is very useful feature and I believe it will save lots of time for future developers! 

Right now this seems to be available only in most recent Windhawk versions. Since the functionality appears to be very new, some users might still not have it (for example, I do not have it in my Windhawk). I do not want to force users to upgrade Windhawk in order to use my mod. Many people might even not understand why compilation errors occur and that these errors are related to the need to upgrade Windhawk. They might just give up and think the mod does not work for them.

Reply:
OK, makes sense.